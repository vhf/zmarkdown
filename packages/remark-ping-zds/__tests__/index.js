import dedent from 'dedent'
import unified from 'unified'
import reParse from 'remark-parse'
import remarkStringify from 'remark-stringify'
import rehypeStringify from 'rehype-stringify'
import remark2rehype from 'remark-rehype'

import plugin from '../src/'


const render = text => unified()
  .use(reParse)
  .use(plugin)
  .use(remark2rehype)
  .use(rehypeStringify)
  .processSync(text)

const fixture = dedent`
  Test ping
  =========

  ping @Clem

  ping @**BALDE SOULEYMANE**

  no ping @dqsjdjsqjhdshqjkhfyhefezhjzjhdsjlfjlsqjdfjhsd

  no ping @**I AM CLEM**
`

test('ping', () => {
  const { contents } = render(fixture)
  expect(contents).toMatchSnapshot()
})

test('ping to markdown', () => {
  const { contents } = unified()
    .use(reParse)
    .use(remarkStringify)
    .use(plugin)
    .processSync(fixture)

  expect(contents).toMatchSnapshot()
})
