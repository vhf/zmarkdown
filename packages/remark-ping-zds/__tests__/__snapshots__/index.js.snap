// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ping 1`] = `
"<h1>Test ping</h1>
<p>ping @Clem</p>
<p>ping @<strong>BALDE SOULEYMANE</strong></p>
<p>no ping @dqsjdjsqjhdshqjkhfyhefezhjzjhdsjlfjlsqjdfjhsd</p>
<p>no ping @<strong>I AM CLEM</strong></p>"
`;

exports[`ping to markdown 1`] = `
"# Test ping

ping @Clem

ping @**BALDE SOULEYMANE**

no ping @dqsjdjsqjhdshqjkhfyhefezhjzjhdsjlfjlsqjdfjhsd

no ping @**I AM CLEM**
"
`;
